pkg/uiserver/uiserver.go

```diff
@@ -53,6 +61,63 @@ func RewriteAssets(fs http.FileSystem, cfg *config.Config, updater UpdateContent
 
 	rewrite("/index.html")
 	rewrite("/diagnoseReport.html")
+
+	overrideDistroAssetsRes(fs, cfg, updater)
+}
+
+func overrideDistroAssetsRes(fs http.FileSystem, cfg *config.Config, updater UpdateContentFunc) {
+	exePath, err := os.Executable()
+	if err != nil {
+		log.Fatal("Failed to get work dir", zap.Error(err))
+	}
+
+	distroResDir := path.Join(path.Dir(exePath), distro.DistroResFolderName)
+	info, err := os.Stat(distroResDir)
+	if err != nil || !info.IsDir() {
+		// just ignore
+		return
+	}
+
+	override := func(assetName string) {
+		assetPath := path.Join("/", distro.DistroResFolderName, assetName)
```

Comment:
If I remember correctly, some distributions will not substitute all files. You may need to be compatible with the behavior.

Reply:
Yep, they just need to supply the files that they want to replace, and we only replace the files they supply.

Reply:
Cool!